engines/optix/OptiXModel.cpp

```diff
@@ -263,6 +279,84 @@ void OptiXModel::_commitMaterials()
         material.second->commit();
 }
 
+bool OptiXModel::_commitSimulationData()
+{
+    if (!_simulationHandler)
+        return false;
+
+    if (!_setIsReadyCallback && !_animationParameters.hasIsReadyCallback())
+    {
+        auto& ap = _animationParameters;
+        ap.setIsReadyCallback(
+            [handler = _simulationHandler] { return handler->isReady(); });
+        ap.setDt(_simulationHandler->getDt(), false);
+        ap.setUnit(_simulationHandler->getUnit(), false);
+        ap.setNumFrames(_simulationHandler->getNbFrames(), false);
+        ap.markModified();
+        _setIsReadyCallback = true;
+    }
+
+    const auto animationFrame = _animationParameters.getFrame();
+
+    if (_tf.initialized &&
+        _simulationHandler->getCurrentFrame() == animationFrame)
+    {
+        return false;
+    }
+
+    auto frameData =
+        static_cast<float*>(_simulationHandler->getFrameData(animationFrame));
+
+    if (!frameData)
+        return false;
+
+    auto context = OptiXContext::get().getOptixContext();
+    const size_t frameSize = _simulationHandler->getFrameSize();
+    setBufferRaw(RT_BUFFER_INPUT, RT_FORMAT_FLOAT, _tf.simulationData,
+                 context["simulation_data"], frameData,
+                 frameSize * sizeof(float));
+
+    return true;
+}
+
+bool OptiXModel::_commitTransferFunction()
+{
+    if (!_transferFunction.isModified() && _tf.initialized)
+        return false;
+
+    auto context = OptiXContext::get().getOptixContext();
+
+    setBuffer(RT_BUFFER_INPUT, RT_FORMAT_FLOAT3, _tf.colors, context["colors"],
+              _transferFunction.getColors());
+
+    setBuffer(RT_BUFFER_INPUT, RT_FORMAT_FLOAT, _tf.opacities,
+              context["opacities"],
+              _transferFunction.calculateInterpolatedOpacities());
+
+    context["value_range"]->setFloat(_transferFunction.getValuesRange().x,
+                                     _transferFunction.getValuesRange().y);
+
+    _transferFunction.resetModified();
+    return true;
+}
+
+bool OptiXModel::commitTransferFunction()
+{
+    const auto dirtyTransferFunction = _commitTransferFunction();
+    const auto dirtySimulationData = _commitSimulationData();
+
+    // Set initialized on first successful commit of both transfer function and
+    // simulation data
+    if (dirtyTransferFunction && dirtySimulationData)
+    {
+        auto context = OptiXContext::get().getOptixContext();
+        context["use_simulation_data"]->setUint(1);
+        _tf.initialized = true;
```

Comment:
It looks like you're missing `context["use_simulation_data"]->setUint(0);` somewhere. Unless it's guaranteed that it cannot be ever the case.

Reply:
It is set when creating the scene.

Reply:
But, is it possible to remove the simulation data once the scene has been created?